# HG changeset patch
# User Michael C. Grant <mcg@cvxr.com>
# Date 1393009232 21600
#      Fri Feb 21 13:00:32 2014 -0600
# Node ID 83963bad5e7d906541c56c8d6a4914ebbf7f1653
# Parent  99454a60bf5e33ba6e5f684fe7b3ad79ea2eb9ea
improvements to Qt & OpenGL linking on the Mac (bug #41574)

* configure.ac: Remove -Wl, prefixes from "-framework Carbon".
* acinclude.m4: Remove -Wl, prefixes from "-framework OpenGL".

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1225,18 +1225,18 @@ fi
 
 ### Check for the Carbon framework on MacOSX systems
 OCTAVE_HAVE_FRAMEWORK([Carbon],
   [[#include <Carbon/Carbon.h>]], [[CGMainDisplayID ()]],
   [have_framework_carbon=yes], [have_framework_carbon=no])
 if test $have_framework_carbon = yes; then
   AC_DEFINE(HAVE_FRAMEWORK_CARBON, 1,
     [Define to 1 if framework CARBON is available.])
-  CARBON_LIBS="-Wl,-framework -Wl,Carbon"
-  AC_MSG_NOTICE([adding -Wl,-framework -Wl,Carbon to CARBON_LIBS])
+  CARBON_LIBS="-framework Carbon"
+  AC_MSG_NOTICE([adding -framework Carbon to CARBON_LIBS])
   AC_SUBST(CARBON_LIBS)
 fi
 
 ### Check for list of libraries needed for OpenGL graphics renderer.
 
 check_opengl=yes
 AC_ARG_WITH([opengl],
   [AS_HELP_STRING([--without-opengl],
diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -1010,18 +1010,18 @@ AC_DEFUN([OCTAVE_CHECK_LIB_OPENGL], [
     ]], [[
     GLint par; glGetIntegerv (GL_VIEWPORT, &par);
     ]],
     have_framework_opengl=yes, have_framework_opengl=no)
 
   if test $have_framework_opengl = yes; then
     AC_DEFINE(HAVE_FRAMEWORK_OPENGL, 1,
       [Define to 1 if framework OPENGL is available.])
-    OPENGL_LIBS="-Wl,-framework -Wl,OpenGL"
-    AC_MSG_NOTICE([adding -Wl,-framework -Wl,OpenGL to OPENGL_LIBS])
+    OPENGL_LIBS="-framework OpenGL"
+    AC_MSG_NOTICE([adding -framework OpenGL to OPENGL_LIBS])
     OCTAVE_CHECK_FUNC_GLUTESSCALLBACK_THREEDOTS
   else
     case $canonical_host_type in
       *-*-mingw32* | *-*-msdosmsvc)
         AC_CHECK_HEADERS([windows.h])
       ;;
     esac
     have_opengl_incs=no
